File: WoofIOS/SwiftPackages/Presentation/Navigation/Sources/PresentationNavigation/WoofNavigator.swift
import DI
import Foundation

@SingleForProtocol
public final class WoofNavigator: NavigatorImplementing {
    private let backStack: WoofBackStack

    public func goTo(_ destination: WoofDestination) {
        backStack.push(destination)
    }

    public func back() {
        backStack.pop()
    }
}

File: WoofAndroid/presentation/navigation/src/main/kotlin/com/perrystreet/woof/presentation/navigation/WoofNavigator.kt
package com.perrystreet.woof.presentation.navigation

import org.koin.core.annotation.Single

@Single
class WoofNavigator(
    private val backStack: WoofBackStack,
) : INavigator {
    override fun goTo(destination: WoofDestination) {
        backStack.push(destination)
    }

    override fun back() {
        backStack.pop()
    }
}
